<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Multivariate Genome Wide Marginal Epistasis Test • mvMAPIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Multivariate Genome Wide Marginal Epistasis Test">
<meta property="og:description" content="Epistasis, commonly defined as the interaction between genetic loci, is known to play an important role in
  the phenotypic variation of complex traits. As a result, many statistical methods have been developed to identify
  genetic variants that are involved in epistasis, and nearly all of these approaches carry out this task by focusing on
  analyzing one trait at a time. Previous studies have shown that jointly modeling multiple phenotypes can often
  dramatically increase statistical power for association mapping. In this study, we present the
  multivariate MArginal ePIstasis Test (mvMAPIT) – a multi-outcome generalization of a recently proposed epistatic
  detection method which seeks to detect marginal epistasis or the combined pairwise interaction effects
  between a given variant and all other variants. By searching for marginal epistatic effects, one can identify genetic
  variants that are involved in epistasis without the need to identify the exact partners with which the variants
  interact – thus, potentially alleviating much of the statistical and computational burden associated with
  conventional explicit search based methods. Our proposed mvMAPIT builds upon this strategy by taking of correlation
  structures between traits to improve the identification of variants involved in epistasis. We formulate mvMAPIT as a
  multivariate linear mixed model and develop a multi-trait variance component estimation algorithm for efficient
  parameter inference and P-value computation. Together with reasonable model approximations, our proposed approach is
  scalable to moderately sized GWA studies. Crawford et al. (2017) &lt;doi:10.1371/journal.pgen.1006869&gt;.">
<meta property="og:image" content="https://lcrawlab.github.io/mvMAPIT/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">mvMAPIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/mvMAPIT.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">Tutorials</li>
    <li>
      <a href="articles/tutorial-docker-mvmapit.html">Dockerized mvMAPIT</a>
    </li>
    <li>
      <a href="articles/tutorial-simulations.html">Simulate Traits</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Studies</li>
    <li>
      <a href="articles/study-phillips-bnabs.html">Synergistic epistasis in binding affinity landscapes</a>
    </li>
    <li>
      <a href="articles/study-wtccc-mice.html">Joint modeling of hematology traits yields epistatic signal in stock of mice</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lcrawlab/mvMAPIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="multivariate-mapit-documentation-">Multivariate MAPIT Documentation <img src="reference/figures/logo.png" align="right" alt="" width="120"><a class="anchor" aria-label="anchor" href="#multivariate-mapit-documentation-"></a>
</h1></div>

<p>Find the full package documentation including examples, and articles here: <a href="https://lcrawlab.github.io/mvMAPIT">Multivariate MAPIT Documentation</a>.</p>
<div class="section level2">
<h2 id="the-multivariate-marginal-epistasis-test-mvmapit">The multivariate MArginal ePIstasis Test (mvMAPIT)<a class="anchor" aria-label="anchor" href="#the-multivariate-marginal-epistasis-test-mvmapit"></a>
</h2>
<p>This R package is a generalization of the <a href="https://github.com/lorinanthony/MAPIT" class="external-link">MAPIT implementation</a> by Crawford et al. (2017)<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> for any number of traits as described by Stamp et al. (2022)<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<p>mvMAPIT is implemented as a set of R and C++ routines, which can be carried out within an R environment.</p>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>Epistasis, commonly defined as the interaction between genetic loci, is known to play an important role in the phenotypic variation of complex traits. As a result, many statistical methods have been developed to identify genetic variants that are involved in epistasis, and nearly all of these approaches carry out this task by focusing on analyzing one trait at a time. However, because of the large combinatorial search space of interactions, most epistasis mapping methods face enormous computational challenges and often suffer from low statistical power.</p>
<p>Previous studies have shown that jointly modeling multiple phenotypes can often dramatically increase statistical power for association mapping. Therefore, here we present the <strong>multivariate MArginal ePIstasis Test (mvMAPIT)</strong> – a multi-outcome generalization of a recently proposed epistatic detection method which seeks to detect <em>marginal epistasis</em> or the combined pairwise interaction effects between a given variant and all other variants. By searching for marginal epistatic effects, one can identify genetic variants that are involved in epistasis without the need to identify the exact partners with which the variants interact – thus, potentially alleviating much of the statistical and computational burden associated with conventional explicit search based methods. Our proposed mvMAPIT builds upon this strategy by taking of correlation structures between traits to improve the identification of variants involved in epistasis. We formulate mvMAPIT as a multivariate linear mixed model and develop a multi-trait variance component estimation algorithm for efficient parameter inference and <em>P</em>-value computation. Together with reasonable model approximations, our proposed approach is scalable to moderately sized GWA studies.</p>
</div>
<div class="section level3">
<h3 id="the-method">The Method<a class="anchor" aria-label="anchor" href="#the-method"></a>
</h3>
<p>The <strong>multivariate MArginal ePIstasis Test</strong> is a multi-outcome extension of the statistical framework MAPIT which aims to identify variants that are involved in epistatic interactions by leveraging the correlation structure of non-additive genetic variation that is shared between multiple traits. The key idea behind the concept of marginal epistasis is to identify variants that are involved in epistasis while avoiding the need to explicitly conduct an exhaustive search over all possible pairwise interactions. As an overview of mvMAPIT and its corresponding software implementation, we will assume that we have access to an GWA study on <code>N</code> individuals denoted as <code>D = {X,Y}</code> where <code>X</code> is an <code>N x J</code> matrix of genotypes with <code>J</code> denoting the number of SNPs (each of which is encoded as <code>{0,1,2}</code> copies of a reference allele at each locus <code>j</code>) and <code>Y</code> denoting a <code>N x D</code> matrix holding <code>D</code> different traits that are measured for each of the <code>N</code> individuals.</p>
<p>The goal of mvMAPIT is to identify variants that have non-zero interaction effects with any other variant in the data. To accomplish this, we examine each SNP in turn and assess the null hypothesis that the variance component is zero. In practice, we use a computationally efficient method of moments algorithm calledMQS to estimate model parameters and to carry out calibrated statistical tests within mvMAPIT.</p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="the-r-environment">The R Environment<a class="anchor" aria-label="anchor" href="#the-r-environment"></a>
</h3>
<p>R is a widely used, free, and open source software environment for statistical computing and graphics. The most recent version of R can be downloaded from the <a href="http://cran.r-project.org/" class="external-link">Comprehensive R Archive Network (CRAN)</a> CRAN provides precompiled binary versions of R for Windows, macOS, and select Linux distributions that are likely sufficient for many users’ needs. Users can also install R from source code; however, this may require a significant amount of effort. For specific details on how to compile, install, and manage R and R-packages, refer to the manual <a href="http://cran.r-project.org/doc/manuals/r-release/R-admin.html" class="external-link">R Installation and Administration</a>.</p>
</div>
<div class="section level3">
<h3 id="r-packages-required-for-mvmapit">R Packages Required for mvMAPIT<a class="anchor" aria-label="anchor" href="#r-packages-required-for-mvmapit"></a>
</h3>
<p>mvMAPIT requires the installation of the following R libraries:</p>
<ul>
<li><a href="https://cran.r-project.org/package=checkmate" class="external-link">checkmate</a></li>
<li><a href="https://cran.r-project.org/package=CompQuadForm" class="external-link">CompQuadForm</a></li>
<li><a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a></li>
<li><a href="https://cran.r-project.org/package=foreach" class="external-link">foreach</a></li>
<li><a href="https://cran.r-project.org/package=harmonicmeanp" class="external-link">harmonicmeanp</a></li>
<li><a href="https://cran.r-project.org/package=logging" class="external-link">logging</a></li>
<li><a href="https://cran.r-project.org/package=mvtnorm" class="external-link">mvtnorm</a></li>
<li><a href="https://cran.r-project.org/package=Rcpp" class="external-link">Rcpp</a></li>
<li><a href="https://cran.r-project.org/package=RcppAlgos" class="external-link">RcppAlgos</a></li>
<li><a href="https://cran.r-project.org/package=RcppArmadillo" class="external-link">RcppArmadillo</a></li>
<li><a href="https://cran.r-project.org/package=RcppParallel" class="external-link">RcppParallel</a></li>
<li><a href="https://cran.r-project.org/package=RcppSpdlog" class="external-link">RcppSpdlog</a></li>
<li><a href="https://cran.r-project.org/package=tidyr" class="external-link">tidyr</a></li>
</ul>
<p>The easiest method to install these packages is with the following example command entered in an R shell:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">install.packages</span>(<span class="kw">c</span>( <span class="st">'checkmate'</span>, </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a>                    <span class="st">'CompQuadForm'</span>, </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a>                    <span class="st">'dplyr'</span>, </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a>                    <span class="st">'foreach'</span>, </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a>                    <span class="st">'harmonicmeanp'</span>, </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a>                    <span class="st">'logging'</span>, </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a>                    <span class="st">'mvtnorm'</span>, </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a>                    <span class="st">'Rcpp'</span>, </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a>                    <span class="st">'RcppAlgos'</span>, </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true"></a>                    <span class="st">'RcppArmadillo'</span>, </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true"></a>                    <span class="st">'RcppParallel'</span>, </span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true"></a>                    <span class="st">'RcppSpdlog'</span>, </span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true"></a>                    <span class="st">'testthat'</span>, </span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true"></a>                    <span class="st">'tidyr'</span>), </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true"></a>                    <span class="dt">dependencies =</span> <span class="ot">TRUE</span>);</span></code></pre></div>
<p>Alternatively, one can also <a href="http://cran.r-project.org/doc/manuals/r-release/R-admin.html#Installing-packages" class="external-link">install R packages from the command-line</a>.</p>
</div>
<div class="section level3">
<h3 id="installing-mvmapit">Installing mvMAPIT<a class="anchor" aria-label="anchor" href="#installing-mvmapit"></a>
</h3>
<p>The easiest way to install the package from sources is to change into the directory of mvMAPIT and run <code>R CMD INSTALL . --preclean</code>. The <code>--preclean</code> flag makes sure that the latest state is run.</p>
</div>
<div class="section level3">
<h3 id="c-functions-required-for-mapit">C++ Functions Required for MAPIT<a class="anchor" aria-label="anchor" href="#c-functions-required-for-mapit"></a>
</h3>
<p>The code in this repository assumes that basic Fortran and C++ libraries and compilers are already set up on the running personal computer or cluster. If not, the mvMAPIT functions and necessary Rcpp packages will not work properly. A simple option is to use <a href="https://gcc.gnu.org/" class="external-link">gcc</a>. macOS users may use this collection by installing the <a href="http://brew.sh/index.html" class="external-link">Homebrew package manager</a> and then typing the following into the terminal:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="ex">brew</span> install gcc</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="openmp">OpenMP<a class="anchor" aria-label="anchor" href="#openmp"></a>
</h3>
<p>Note that mvMAPIT takes advantage of <a href="http://openmp.org/wp/" class="external-link">OpenMP</a>, an API for multi-platform shared-memory parallel programming in C/C++. This is to speed up the computational time of the modeling algorithm. Unfortunately, macOS does not currently support OpenMP under the default compiler. A work around to use OpenMP in R on macOS can be found <a href="http://thecoatlessprofessor.com/programming/openmp-in-r-on-os-x/" class="external-link">here</a>. mvMAPIT can be compiled without OpenMP, but we recommend using it if applicable.</p>
</div>
<div class="section level3">
<h3 id="known-issues">Known Issues<a class="anchor" aria-label="anchor" href="#known-issues"></a>
</h3>
<ul>
<li><p>When your compiler changes, some R package dependencies might need to be recompiled. This is likely the case if the compilation error explicitly names an R package in the local library.</p></li>
<li><p>On macOS, you might need to run <code>brew reinstall z3</code> to fix <code>'libz3.4.11.dylib' (no such file)</code> related errors (<a href="https://github.com/Homebrew/discussions/discussions/3920" class="external-link">clang issues</a>).</p></li>
<li><p>For extra tips on how to run C++ on macOS, please visit <a href="http://seananderson.ca/2013/11/18/rcpp-mavericks.html" class="external-link uri">http://seananderson.ca/2013/11/18/rcpp-mavericks.html</a>.</p></li>
<li><p>For tips on how to avoid errors dealing with <code>-lgfortran</code> or <code>-lquadmath</code>, please visit <a href="http://thecoatlessprofessor.com/programming/rcpp-rcpparmadillo-and-os-x-mavericks-lgfortran-and-lquadmath-error/" class="external-link uri">http://thecoatlessprofessor.com/programming/rcpp-rcpparmadillo-and-os-x-mavericks-lgfortran-and-lquadmath-error/</a>.</p></li>
</ul>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="questions-and-feedback">Questions and Feedback<a class="anchor" aria-label="anchor" href="#questions-and-feedback"></a>
</h2>
<p>For questions or concerns with the MAPIT functions, please contact <a href="mailto:lorin_crawford@brown.edu">Lorin Crawford</a> or <a href="mailto:julian_stamp@brown.edu">Julian Stamp</a>.</p>
<p>We appreciate any feedback you may have with our repository and instructions.</p>
</div>
<div class="section level2"><h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2></div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>L. Crawford, P. Zeng, S. Mukherjee, X. Zhou (2017). Detecting epistasis with the marginal epistasis test in genetic mapping studies of quantitative traits. <em>PLoS Genet</em>. <strong>13</strong> (7): e1006869. <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006869" class="external-link uri">http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006869</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>J. Stamp, A. DenAdel, D. Weinreich, L. Crawford (2022). Multivariate MAPIT.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcrawlab/mvMAPIT/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing mvMAPIT</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Julian Stamp <br><small class="roles"> Maintainer, author </small>  </li>
<li>Lorin Crawford <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcrawlab/mvMAPIT/actions/workflows/check-standard.yaml" class="external-link"><img src="https://github.com/lcrawlab/mvMAPIT/actions/workflows/check-standard.yaml/badge.svg" alt="R CMD check"></a></li>
<li><a href="https://github.com/lcrawlab/mvMAPIT/actions/workflows/docker-image.yml" class="external-link"><img src="https://github.com/lcrawlab/mvMAPIT/actions/workflows/docker-image.yml/badge.svg" alt="Docker Image CI"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
