---
title: "Illustrating the multivariate MArginal ePIstasis Test (MAPIT) with Simulations"
author: "Julian Stamp"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Illustrating the multivariate MArginal ePIstasis Test (MAPIT) with Simulations}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
library(knitr)
```

# Standard workflow

> **_NOTE:_** if you use MAPIT in published research, please cite:
> placeholder

## Getting Started

Load necessary libraries

> **_NOTE:_**
This code assumes that the basic C++ functions are set up on the computer in use.
> The `Rcpp` packages rely on C++ libraries, e.g. `openblas`, `lapack`, `flibs`.
> Mac users please refer to the homebrew applications and install the `gcc` commands listed in the `README.md` file.

```{r load_dependencies, eval = FALSE}
library(mvMAPIT)
```
## Simulating Data

Data are $p$ single nucleotide polymorphisms (SNPs) and $n$ simulated sample genotypes.

Simulation Parameters:

1. `n`: number of samples
1. `p`: number of SNPs or variants
1. `d`: number of traits
1. `PVE`: phenotypic variance explained/broad-sense heritability ($H^2$)
1. `rho`: measures the portion of $H^2$ that is contributed by the marignal (additive) effects
1. Number of SNPs that are causing variation in the trait:
    * `n_causal`: total number of SNPs with any effects on traits.
    * `n_trait_specific`: number of epistatic SNPs that affect a single trait.
    * `n_pleiotropic`: number of epistatic SNPs that affect all traits.
    * `group_ratio_pleiotropic`: ratio by which to split the epistatic SNPs into groups.
1. `epistatic_correlation`: correlation between the epistatic effects.

```{r simulation_parameters, eval = FALSE}
seed <- 67132
n <- 100
p <- 2000
d <- 2
PVE <- 0.6
rho <- 0.8
n_causal <- 1000
n_trait_specific <- 10
n_pleiotropic <- 10
group_ratio_pleiotropic <- 1
epistatic_correlation <- 0.9
maf <- 0.05
X <- matrix(
    runif(p * n),
    ncol = p
)
data <- simulate_traits(
    X,
    n_causal = n_causal,
    n_trait_specific = n_trait_specific,
    n_pleiotropic = n_pleiotropic,
    d = d,
    H2 = PVE,
    rho = rho,
    epistatic_correlation = epistatic_correlation,
    group_ratio_pleiotropic = group_ratio_pleiotropic,
    maf_threshold = maf,
    seed = seed,
    logLevel = "ERROR"
)
```
The entries in the genotype matrix are:

* $0$ if the genotype is homozygous in the reference allele,
* $1$ if both, the reference allele and the alternative allele are present,
* $2$ if the genotype is homozygous in the alternative allele.

An example genotype matrix could look like this:

|Sample|SNP 1|SNP 2|...|SNP p|
|------|-----|-----|---|-----|
|  1   |  0  |  1  | 2 |  2  |
|  2   |  1  |  0  | 2 |  0  |
| ...  |  0  |  1  | 0 |  1  |
|  n   |  2  |  2  | 1 |  1  |

As in the paper, the genotype matrix will be scaled and centered such that the data at every SNP has mean $\mu = 0$ and standard deviation $\sigma = 1$. The genotypes are simulated such that all variants have minor allele frequency $\mathrm{MAF} > 0.05$. For each SNP a minor allele frequency is drawn at random from a uniform distribution $\mathrm{MAF} \sim \mathrm{Uniform}(0.05, 0.5)$. Then, haplotypes for each SNP are simulated by drawing from a $\mathrm{Uniform}(0, 1)$ distribution and comparing the drawn value to the minor allele frequency. The values less than the minor allele frequency are mapped to 1, the ones above to zero. The genotype is simulated by adding the haplotype vectors.

As in the paper, we randomly choose causal variants that classify into three groups:

1. set 1 of interaction SNPs,
1. set 2 of interaction SNPs, and
1. a large set of additive SNPs.

In the simulations carried out in this study, SNPs interact between sets. SNPs in the first group interact with SNPs in the second group, but do not interact with variants in their own group, and vice versa. One may view the SNPs in the first set as the “hubs” in an interaction map. We are reminded that interaction (epistatic) effects are different from additive effects.

> **NOTE**: All causal SNPs in both the first and second groups have additive effects and are involved in pairwise interactions, while causal SNPs in the third set only have additive effects.

To summarize, each effect size for the marginal, epistatic, and random error effects are drawn from a standard normal distribution. Meaning $\beta$ ~ MVN(0,I), $\alpha$ ~ MVN(0,I), and $\epsilon$ ~ MVN(0,I). We then scale both the additive and pairwise genetic effects so that collectively they explain a fixed proportion of genetic variance. Namely, the additive effects make up $\rho$%, while the pairwise interactions make up the remaining $(1 − \rho)$% of the explained heritability. Once we obtain the final effect sizes for all causal SNPs, we draw errors to achieve the target $H^2$ such that

## Running mvMAPIT

:warning: mvMAPIT takes the X matrix as $p \times n$; **not** as  $n \times p$.

Set the number of cores
```{r, eval = FALSE}
cores <- detectCores()
```
mvMAPIT can be run in multiple modes. By default it runs in a hybrid mode, performing tests both wtih a normal Z-test as well as the Davies method.
```{r run_mapit_normal, eval = FALSE}
mapit <- mvmapit(
        t(data$genotype),
        t(data$trait),
        test = "normal"
)
```

```{r load_mapit_analysis_data, include=FALSE}
load('../data/mapit_analysis_data.rda') # knitr is run in the vignettes directory
hybrid.pvals <- mapit_analysis_data$hybrid_pvalues
normal.pvals <- mapit_analysis_data$normal_pvalues
davies.pvals <- mapit_analysis_data$davies_pvalues
epistatic_snps <- mapit_analysis_data$epistatic_snps
additive_snps <- mapit_analysis_data$additive_snps
exhaustive_search_results <- mapit_analysis_data$exhaustive_search
```

### Running an Informed Exhaustive Search

> **NOTE**: Now we may take only the significant SNPs according to their marginal epistatic effects and run a simple exhaustive search between them

```{r search_significant_SNPs, eval = FALSE}
thresh <- 0.05 / length(mapit$pvalues) # Set a significance threshold
significant_snps <- hybrid.pvals[hybrid.pvals <= thresh] # Call only marginally significant SNPs
significant_snps <- significant_snps[!is.na(significant_snps)]
pairs <- c()
if (length(significant_snps) > 1) {
  pairnames <- comboGeneral(names(significant_snps), 2) # Generate unique pairs of SNP names; for length(names) = n, the result is a (n * (n-1)) x 2 matrix with one row corresponding to a pair
  for (k in seq_len(nrow(pairnames))) {
    fit <- lm(y ~ X[, pairnames[k, 1]]:X[, pairnames[k, 2]])
    p_value <- coefficients(summary(fit))[8]
    names(p_value) <- paste(pairnames[k, 1], pairnames[k, 2], sep = "-")
    pairs <- c(pairs, p_value)
  }
}
```

