<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Simulate phenotye data — simulate_traits • mvMAPIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Simulate phenotye data — simulate_traits"><meta property="og:description" content="This function simulates trait data from a genotype matrix."><meta property="og:image" content="https://lcrawlab.github.io/mvMAPIT/logo.png"><meta name="google-site-verification" content="7EMmOVQ0-Feh--5i1ogdQqVYz7Azip5Je-UeBYXP6fA"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mvMAPIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mvMAPIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="divider">
    <li class="dropdown-header">Tutorials</li>
    <li>
      <a href="../articles/tutorial-docker-mvmapit.html">Dockerized mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-simulations.html">Simulate Traits</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Studies</li>
    <li>
      <a href="../articles/study-compare-p-value-combine-methods.html">Empirical comparison of P-value combination methods in mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/study-phillips-bnabs.html">Synergistic epistasis in binding affinity landscapes</a>
    </li>
    <li>
      <a href="../articles/study-wtccc-mice.html">Joint modeling of hematology traits yields epistatic signal in stock of mice</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/lcrawlab/mvMAPIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Simulate phenotye data</h1>
    <small class="dont-index">Source: <a href="https://github.com/lcrawlab/mvMAPIT/blob/HEAD/R/simulate_data.R" class="external-link"><code>R/simulate_data.R</code></a></small>
    <div class="hidden name"><code>simulate_traits.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function simulates trait data from a genotype matrix.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">simulate_traits</span><span class="op">(</span></span>
<span>  <span class="va">genotype_matrix</span>,</span>
<span>  n_causal <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  n_trait_specific <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  n_pleiotropic <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  H2 <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  d <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  marginal_correlation <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  epistatic_correlation <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  group_ratio_trait <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  group_ratio_pleiotropic <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  maf_threshold <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">67132</span>,</span>
<span>  logLevel <span class="op">=</span> <span class="st">"INFO"</span>,</span>
<span>  logFile <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genotype_matrix</dt>
<dd><p>Genotype matrix with samples as rows, and SNPs as columns.</p></dd>


<dt>n_causal</dt>
<dd><p>Number of SNPs that are causal.</p></dd>


<dt>n_trait_specific</dt>
<dd><p>Number of causal SNPs with single trait epistatic effects.</p></dd>


<dt>n_pleiotropic</dt>
<dd><p>Number of SNPs with epistatic effects on all traits.</p></dd>


<dt>H2</dt>
<dd><p>Broad-sense heritability. Can be vector.</p></dd>


<dt>d</dt>
<dd><p>Number of traits.</p></dd>


<dt>rho</dt>
<dd><p>Proportion of heritability explained by additivity.</p></dd>


<dt>marginal_correlation</dt>
<dd><p>Correlation between the additive effects of the trait.</p></dd>


<dt>epistatic_correlation</dt>
<dd><p>Correlation between the epistatic effects of the trait.</p></dd>


<dt>group_ratio_trait</dt>
<dd><p>Ratio of sizes of trait specific groups that interact, e.g. a ratio 1:3 would be value 3.</p></dd>


<dt>group_ratio_pleiotropic</dt>
<dd><p>Ratio of sizes of pleiotropic groups that interact, e.g. a ratio 1:3 would be value 3.</p></dd>


<dt>maf_threshold</dt>
<dd><p>is a float parameter defining the threshold for the minor allele frequency not included in causal SNPs.</p></dd>


<dt>seed</dt>
<dd><p>Random seed for simulation.</p></dd>


<dt>logLevel</dt>
<dd><p>is a string parameter defining the log level for the logging package.</p></dd>


<dt>logFile</dt>
<dd><p>is a string parameter defining the name of the log file for the logging output.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list object containing the trait data, the genotype data, as well as the causal SNPs and summary statistics.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function takes a genotype matrix and simulates trait data under the following model:
beta_i ~ MN(0, V_i, I), i in { additive, epistatic, residual}</p>
<p>The effect sizes follow a matrix normal distribution with no correlation between the samples but covariance between the effects for different traits</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fl">200</span></span></span>
<span class="r-in"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fl">10</span></span></span>
<span class="r-in"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fl">4</span></span></span>
<span class="r-in"><span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">100</span></span></span>
<span class="r-in"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">p</span> <span class="op">*</span> <span class="va">n</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    ncol <span class="op">=</span> <span class="va">p</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">simulate_traits</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">X</span>, n_causal <span class="op">=</span> <span class="va">f</span>, n_trait_specific <span class="op">=</span> <span class="va">g</span>, n_pleiotropic <span class="op">=</span> <span class="va">g</span>, d <span class="op">=</span> <span class="va">d</span>, maf_threshold <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>    logLevel <span class="op">=</span> <span class="st">"ERROR"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

