<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Joint modeling of hematology traits yields epistatic signal in stock of mice • mvMAPIT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Joint modeling of hematology traits yields epistatic signal in stock of mice">
<meta property="og:description" content="Example application of mvMAPIT to hematology traits in mice.
">
<meta property="og:image" content="https://lcrawlab.github.io/mvMAPIT/logo.png">
<meta name="google-site-verification" content="7EMmOVQ0-Feh--5i1ogdQqVYz7Azip5Je-UeBYXP6fA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mvMAPIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/mvMAPIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">Tutorials</li>
    <li>
      <a href="../articles/tutorial-docker-mvmapit.html">Dockerized mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-lt-mapit.html">Liability threshold MAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-simulations.html">Simulate Traits</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Studies</li>
    <li>
      <a href="../articles/study-compare-p-value-combine-methods.html">Empirical comparison of P-value combination methods in mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/study-phillips-bnabs.html">Synergistic epistasis in binding affinity landscapes</a>
    </li>
    <li>
      <a href="../articles/study-wtccc-mice.html">Joint modeling of hematology traits yields epistatic signal in stock of mice</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lcrawlab/mvMAPIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Joint modeling of hematology traits yields
epistatic signal in stock of mice</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/mvMAPIT/blob/main/vignettes/study-wtccc-mice.Rmd" class="external-link"><code>vignettes/study-wtccc-mice.Rmd</code></a></small>
      <div class="hidden name"><code>study-wtccc-mice.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/mvMAPIT" class="external-link">mvMAPIT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<blockquote>
<p><strong>NOTE:</strong> Due to the size of the data, this vignette
does not contain the output of the code.</p>
</blockquote>
<div class="section level2">
<h2 id="preprocessing-of-the-heterogenous-stock-of-mice-dataset">Preprocessing of the heterogenous stock of mice dataset<a class="anchor" aria-label="anchor" href="#preprocessing-of-the-heterogenous-stock-of-mice-dataset"></a>
</h2>
<p>This example study makes use of GWA data from the Wellcome Trust
Centre for Human
Genetics<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>3</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2,3}</annotation></semantics></math>
(<a href="http://mtweb.cs.ucl.ac.uk/mus/www/mouse/index.shtml" class="external-link uri">http://mtweb.cs.ucl.ac.uk/mus/www/mouse/index.shtml</a>).
The genotypes from this study were downloaded directly using the
<code>BGLR-R</code> package. This study contains
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>N</mi><mo>=</mo></mrow><annotation encoding="application/x-tex">N =</annotation></semantics></math>
1,814 heterogenous stock of mice from 85 families (all descending from
eight inbred progenitor
strains)<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2}</annotation></semantics></math>,
and 131 quantitative traits that are classified into 6 broad categories
including behavior, diabetes, asthma, immunology, haematology, and
biochemistry. Phenotypic measurements for these mice can be found freely
available online to download (details can be found at <a href="http://mtweb.cs.ucl.ac.uk/mus/www/mouse/HS/index.shtml" class="external-link uri">http://mtweb.cs.ucl.ac.uk/mus/www/mouse/HS/index.shtml</a>).
In the main text, we focused on 15 hematological phenotypes including:
atypical lymphocytes (ALY; <code>Haem.ALYabs</code>), basophils (BAS;
<code>Haem.BASabs</code>), hematocrit (HCT; <code>Haem.HCT</code>),
hemoglobin (HGB; <code>Haem.HGB</code>), large immature cells (LIC;
<code>Haem.LICabs</code>), lymphocytes (LYM; <code>Haem.LYMabs</code>),
mean corpuscular hemoglobin (MCH; <code>Haem.MCH</code>), mean
corpuscular volume (MCV; <code>Haem.MCV</code>), monocytes (MON;
<code>Haem.MONabs</code>), mean platelet volume (MPV;
<code>Haem.MPV</code>), neutrophils (NEU; <code>Haem.NEUabs</code>),
plateletcrit (PCT; <code>Haem.PCT</code>), platelets (PLT;
<code>Haem.PLT</code>), red blood cell count (RBC;
<code>Haem.RBC</code>), red cell distribution width (RDW;
<code>Haem.RDW</code>), and white blood cell count (WBC;
<code>Haem.WBC</code>). All phenotypes were previously corrected for
sex, age, body weight, season, year, and cage effects
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2}</annotation></semantics></math>.
For individuals with missing genotypes, we imputed values by the mean
genotype of that SNP in their corresponding family. Only polymorphic
SNPs with minor allele frequency above 5% were kept for the analyses.
This left a total of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>J</mi><mo>=</mo></mrow><annotation encoding="application/x-tex">J =</annotation></semantics></math>
10,227 autosomal SNPs that were available for all mice.</p>
</div>
<div class="section level2">
<h2 id="analyze-hematology-traits-in-mice">Analyze hematology traits in mice<a class="anchor" aria-label="anchor" href="#analyze-hematology-traits-in-mice"></a>
</h2>
<p>In this section, we apply mvMAPIT to individual-level genotypes and
15 hematology traits in a heterogeneous stock of mice dataset from the
Wellcome Trust Centre for Human
Genetics<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>3</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2,3}</annotation></semantics></math>.
This collection of data contains approximately
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>N</mi><mo>=</mo></mrow><annotation encoding="application/x-tex">N =</annotation></semantics></math>
2,000 individuals depending on the phenotype, and each mouse has been
genotyped at
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>J</mi><mo>=</mo></mrow><annotation encoding="application/x-tex">J =</annotation></semantics></math>
10,346 SNPs. Specifically, this stock of mice are known to be
genetically related with population structure and the genetic
architectures of these particular traits have been shown to have
different levels of broad-sense heritability with varying contributions
from non-additive genetic effects.</p>
<div class="section level3">
<h3 id="apply-mvmapit">Apply mvMAPIT<a class="anchor" aria-label="anchor" href="#apply-mvmapit"></a>
</h3>
<p>The number of complete samples in the data varies for different
traits and trait pairs. For this study we created separate data sets for
each trait and trait pair containing the genotype data in a genotype
matrix encoded as <code>{0, 1, 2}</code> (minor allele count) and the
trait or trait pair in a phenotype matrix. Apply <code><a href="../reference/mvmapit.html">mvmapit()</a></code>
to each data set by running the following.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mvmapit_TRAIT</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mvmapit.html">mvmapit</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">TRAIT</span><span class="op">$</span><span class="va">genotype</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">TRAIT</span><span class="op">$</span><span class="va">phenotype</span><span class="op">)</span>,</span>
<span>  test <span class="op">=</span> <span class="st">"hybrid"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>As a result, we get redundant
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>-values
for some of the univariate variance components. The statistical
detection of epistasis is sensitive to sample size. Therefore, we
coalesce the redundant data by keeping the analysis results of the
largest data set used in the analysis and impute missing data from the
next smaller data set that has no missing data.</p>
</div>
<div class="section level3">
<h3 id="analysis-data-availability">Analysis Data Availability<a class="anchor" aria-label="anchor" href="#analysis-data-availability"></a>
</h3>
<p>The results of the paper data are published on Harvard Dataverse.
Find the files for Download <a href="https://doi.org/10.7910/DVN/WPFIGU" class="external-link">here</a><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>23</mn></msup><annotation encoding="application/x-tex">^{23}</annotation></semantics></math>.
For running the code snippets in this vignette, download the two
files</p>
<ul>
<li>mice_HCTHGB_MCVMCH.rds</li>
<li>mice_SI_paper.rds</li>
</ul>
<p>and read the files using the following:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mice_SI_paper</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"mice_SI_paper.rds"</span><span class="op">)</span></span>
<span><span class="va">mice_HCTHGB_MCVMCH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"mice_HCTHGB_MCVMCH.rds"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="all-traits-overview">All Traits Overview<a class="anchor" aria-label="anchor" href="#all-traits-overview"></a>
</h3>
<p>We also include results corresponding to the univariate MAPIT model
and the covariance test for comparison. Overall, the single-trait
marginal epistatic test does only identifies significant variants for
the large immature cells (LIC) after Bonferroni correction
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>4.83</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>6</mn></mrow></msup></mrow><annotation encoding="application/x-tex">P = 4.83\times 10^{-6}</annotation></semantics></math>).
A complete picture of this can be seen in the following figure, which
depicts Manhattan plots of our genome-wide interaction study for all
combinations of trait pairs. Here, we can see that most of the signal in
the combined
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>-values
from mvMAPIT likely stems from the covariance component portion of the
model.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">for_ticks_chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">position</span> <span class="op">~</span> <span class="va">chr</span>, <span class="va">mice_data</span><span class="op">$</span><span class="va">fisher</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>first <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, last <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>tick <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a></span><span class="op">(</span><span class="op">(</span><span class="va">position</span><span class="op">[</span>,<span class="st">"first"</span><span class="op">]</span> <span class="op">+</span> <span class="va">position</span><span class="op">[</span>,<span class="st">"last"</span><span class="op">]</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>chr2 <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case-and-replace-when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">5</span> <span class="op">==</span> <span class="fl">0</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">chr</span><span class="op">)</span>,</span>
<span>                          <span class="va">chr</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">chr</span><span class="op">)</span>,</span>
<span>                          <span class="cn">TRUE</span> <span class="op">~</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">for_facetgrid_row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/as_labeller.html" class="external-link">as_labeller</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`1` <span class="op">=</span> <span class="st">"Trait #1"</span>, `2` <span class="op">=</span> <span class="st">"Trait #2"</span>, `3` <span class="op">=</span> <span class="st">"Covariance"</span>, `4` <span class="op">=</span> <span class="st">"Combined"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">mice_SI_paper</span><span class="op">$</span><span class="va">fisher</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="va">position</span>,</span>
<span>      y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pplot</span><span class="op">)</span>,</span>
<span>      color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">color</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu">geom_point_rast</span><span class="op">(</span></span>
<span>        size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span></span>
<span>        values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#8b8b8b"</span>, <span class="st">"#bfbfbf"</span>, <span class="st">"#1b9e77"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span>,</span>
<span>                         labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"0"</span>, <span class="st">"5"</span>, <span class="st">"&gt;10"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>          yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">5.179737e-06</span><span class="op">)</span>,</span>
<span>          linetype <span class="op">=</span> <span class="st">"Bonferroni"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="st">"#d95f02"</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">0.3</span></span>
<span>      <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">x</span> <span class="op">~</span> <span class="va">y</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"Times"</span><span class="op">)</span>,</span>
<span>      <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>        y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/bquote.html" class="external-link">bquote</a></span><span class="op">(</span><span class="op">-</span><span class="va">log</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span><span class="st">"Chromosome"</span>,</span>
<span>                         breaks <span class="op">=</span> <span class="va">for_ticks_chr</span><span class="op">$</span><span class="va">tick</span>,</span>
<span>                         labels <span class="op">=</span> <span class="va">for_ticks_chr</span><span class="op">$</span><span class="va">chr2</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_linetype_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">""</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'dashed'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="two-trait-pairs-hct-hgb-as-well-as-mcv-mch">Two trait pairs HCT &amp; HGB as well as MCV &amp; MCH<a class="anchor" aria-label="anchor" href="#two-trait-pairs-hct-hgb-as-well-as-mcv-mch"></a>
</h3>
<p>The hypothesis that most of the signal in the combined
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>-values
from mvMAPIT likely stems from the covariance component portion of the
model holds true for the joint pairwise analysis of hematocrit (HCT) and
hemoglobin (HGB) and mean corpuscular hemoglobin (MCH) and mean
corpuscular volume (MCV) (e.g., see the third and fourth rows of the
following figure).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">mice_HCTHGB_MCVMCH</span><span class="op">$</span><span class="va">fisher</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="va">position</span>,</span>
<span>      y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pplot</span><span class="op">)</span>,</span>
<span>      color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">color</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu">geom_point_rast</span><span class="op">(</span></span>
<span>        size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span></span>
<span>        values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#8b8b8b"</span>, <span class="st">"#bfbfbf"</span>, <span class="st">"#1b9e77"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span>,</span>
<span>                         labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"0"</span>, <span class="st">"5"</span>, <span class="st">"&gt;10"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>          yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">5.179737e-06</span><span class="op">)</span>,</span>
<span>          linetype <span class="op">=</span> <span class="st">"Bonferroni"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="st">"#d95f02"</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">0.3</span></span>
<span>      <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">row</span> <span class="op">~</span> <span class="va">case</span>, labeller <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labeller.html" class="external-link">labeller</a></span><span class="op">(</span>row <span class="op">=</span> <span class="va">for_facetgrid_row</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"Times"</span><span class="op">)</span>,</span>
<span>      <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>        y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/bquote.html" class="external-link">bquote</a></span><span class="op">(</span><span class="op">-</span><span class="va">log</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span><span class="st">"Chromosome"</span>,</span>
<span>                         breaks <span class="op">=</span> <span class="va">for_ticks_chr</span><span class="op">$</span><span class="va">tick</span>,</span>
<span>                         labels <span class="op">=</span> <span class="va">for_ticks_chr</span><span class="op">$</span><span class="va">chr2</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_linetype_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">""</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'dashed'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="va">gg</span><span class="op">)</span></span></code></pre></div>
<p>One explanation for observing more signal in the covariance
components over the univariate test could be derived from the traits
having low heritability but high correlation between epistatic
interaction effects. In our simulation studies (see publication) we
showed that the sensitivity of the covariance statistic increased for
these cases. Notably, the non-additive signal identified by the
covariance test is not totally dependent on the empirical correlation
between traits. Instead, as previously shown in our simulation study,
the power of mvMAPIT over the univariate approach occurs when there is
correlation between the effects of epistatic interactions shared between
two traits. Importantly, many of the candidate SNPs selected by the
mvMAPIT framework have been previously discovered by past publications
as having some functional nonlinear relationship with the traits of
interest. For example, the multivariate analysis with traits MCH and MCV
show a significant SNP rs4173870
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>4.89</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>10</mn></mrow></msup></mrow><annotation encoding="application/x-tex">P = 4.89\times 10^{-10}</annotation></semantics></math>)
in the gene hematopoietic cell-specific Lyn substrate 1 (Hcls1) on
chromosome 16 which has been shown to play a role in differentiation of
erythrocytes<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>7</mn></msup><annotation encoding="application/x-tex">^7</annotation></semantics></math>.
Similarly, the joint analysis of HGB and HCT shows hits in multiple
coding regions. One example here are the SNPs rs3692165
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>1.82</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>6</mn></mrow></msup></mrow><annotation encoding="application/x-tex">P = 1.82\times 10^{-6}</annotation></semantics></math>)
and rs13482117
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>8.94</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>7</mn></mrow></msup></mrow><annotation encoding="application/x-tex">P = 8.94\times 10^{-7}</annotation></semantics></math>)
in the gene calcium voltage-gated channel auxiliary subunit alpha2delta
3 (Cacna2d3) on chromosome 14, which has been associated with decreased
circulating glucose
levels<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math>,
and SNP rs3724260
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>4.58</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>6</mn></mrow></msup></mrow><annotation encoding="application/x-tex">P = 4.58\times 10^{-6}</annotation></semantics></math>)
in the gene Dicer1 on chromosome 12 which has been annotated for anemia
both in humans and
mice<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>9</mn></msup><annotation encoding="application/x-tex">^9</annotation></semantics></math>.</p>
</div>
<div class="section level3">
<h3 id="notable-snps-with-marginal-epistatic-effects-after-applying-the-mvmapit-framework-to-15-hematology-traits">Notable SNPs with marginal epistatic effects after applying the
mvMAPIT framework to 15 hematology traits<a class="anchor" aria-label="anchor" href="#notable-snps-with-marginal-epistatic-effects-after-applying-the-mvmapit-framework-to-15-hematology-traits"></a>
</h3>
<p>For full analysis, we provide a summary table which lists the
combined
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>-values
after running mvMAPIT with Fisher’s method. The following table lists a
select subset of SNPs in coding regions of genes that have been
associated with phenotypes related to the hematopoietic system, immune
system, or homeostasis and metabolism. Each of these are significant
(after correction for multiple hypothesis testing) in the mvMAPIT
analysis of related hematology traits. Some of these phenotypes have
been reported as having large broad-sense heritability, which improves
the ability of mvMAPIT to detect the signal. For example, the genes Arf2
and Cacna2d3 are associated with phenotypes related to glucose
homeostasis, which has been reported to have a large heritable component
(estimated
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>H</mi><mn>2</mn></msup><mo>=</mo><mn>0.3</mn></mrow><annotation encoding="application/x-tex">H^2 = 0.3</annotation></semantics></math>
for insulin
sensitivity<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>10</mn></msup><annotation encoding="application/x-tex">^{10}</annotation></semantics></math>).
Similarly, the genes App and Pex1 are associated with thrombosis where
(an estimated) more than half of phenotypic variation has been
attributed to genetic effects (estimated
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>H</mi><mn>2</mn></msup><mo>≥</mo><mn>0.6</mn></mrow><annotation encoding="application/x-tex">H^2 \ge 0.6</annotation></semantics></math>
for susceptibility to common
thrombosis<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>11</mn></msup><annotation encoding="application/x-tex">^{11}</annotation></semantics></math>).</p>
<table class="table">
<colgroup>
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th><strong>SNP</strong></th>
<th><strong>Location</strong></th>
<th><strong>Trait 1</strong></th>
<th><strong>Trait 2</strong></th>
<th><strong>Trait 1
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math></strong></th>
<th><strong>Trait 2
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math></strong></th>
<th><strong>Cov.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math></strong></th>
<th><strong>Comb.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math></strong></th>
<th><strong>Gene</strong></th>
<th><strong>Genomic Annotation</strong></th>
<th><strong>Reference</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>rs3699393</td>
<td>2:5887012</td>
<td>MCV</td>
<td>PLT</td>
<td>0.21</td>
<td>0.23</td>
<td>5.75e-7</td>
<td>4.9e-06</td>
<td>Upf2</td>
<td>anemia and abnormal bone marrow cell development</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>12</mn></msup><annotation encoding="application/x-tex">^{12}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs13478092</td>
<td>5:3601413</td>
<td>LIC</td>
<td>PLT</td>
<td>0.034</td>
<td>0.58</td>
<td>1.67e-10</td>
<td>1.26e-9</td>
<td>Pex1</td>
<td>abnormal venous thrombosis</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>13</mn></msup><annotation encoding="application/x-tex">^{13}</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs3694887</td>
<td>5:102770070</td>
<td>ALY</td>
<td>LIC</td>
<td>1.26e-4</td>
<td>0.013</td>
<td>2.54e-6</td>
<td>1.55e-9</td>
<td>Aff1</td>
<td>abnormal B and T cell number and morphology</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>14</mn></msup><annotation encoding="application/x-tex">^{14}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs3694887</td>
<td>5:102770070</td>
<td>LIC</td>
<td>PLT</td>
<td>0.013</td>
<td>0.28</td>
<td>5.47e-27</td>
<td>4.49e-26</td>
<td>Aff1</td>
<td>abnormal B and T cell number and morphology</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>14</mn></msup><annotation encoding="application/x-tex">^{14}</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs13478923</td>
<td>6:99475169</td>
<td>ALY</td>
<td>LIC</td>
<td>2.8e-4</td>
<td>0.12</td>
<td>1.79e-6</td>
<td>1.81e-8</td>
<td>Foxp1</td>
<td>abnormal B cell differentiation, physiology, count</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>15</mn><mo>,</mo><mn>16</mn></mrow></msup><annotation encoding="application/x-tex">^{15,16}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs13478924</td>
<td>6:99571626</td>
<td>ALY</td>
<td>LIC</td>
<td>3.11e-4</td>
<td>0.12</td>
<td>2.70e-6</td>
<td>2.86e-8</td>
<td>Foxp1</td>
<td>abnormal B cell differentiation, physiology, count</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>15</mn><mo>,</mo><mn>16</mn></mrow></msup><annotation encoding="application/x-tex">^{15,16}</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs13478985</td>
<td>6:115245823</td>
<td>MCV</td>
<td>WBC</td>
<td>0.16</td>
<td>0.40</td>
<td>1.14e-81</td>
<td>1.34e-78</td>
<td>Atg7</td>
<td>decreased bone marrow cell count</td>
<td>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>17</mn></msup><annotation encoding="application/x-tex">^{17}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>18</mn></msup><annotation encoding="application/x-tex">^{18}</annotation></semantics></math>
</td>
</tr>
<tr class="even">
<td>rs3723163</td>
<td>11:103800737</td>
<td>HCT</td>
<td>LYM</td>
<td>0.072</td>
<td>0.30</td>
<td>3.99e-107</td>
<td>2.66e-104</td>
<td>Arf2</td>
<td>decreased fasting circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs3723163</td>
<td>11:103800737</td>
<td>HGB</td>
<td>WBC</td>
<td>0.069</td>
<td>0.25</td>
<td>1.85e-7</td>
<td>6.76e-7</td>
<td>Arf2</td>
<td>decreased fasting circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs3724260</td>
<td>12:100163212</td>
<td>HGB</td>
<td>HCT</td>
<td>0.030</td>
<td>0.062</td>
<td>1.44e-5</td>
<td>4.58e-6</td>
<td>Dicer1</td>
<td>anemia</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>9</mn></msup><annotation encoding="application/x-tex">^9</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs3692165</td>
<td>14:27756640</td>
<td>HCT</td>
<td>HGB</td>
<td>0.026</td>
<td>0.037</td>
<td>9.9e-6</td>
<td>1.8e-06</td>
<td>Cacna2d3</td>
<td>decreased circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs3697466</td>
<td>14:27485228</td>
<td>HCT</td>
<td>HGB</td>
<td>0.026</td>
<td>0.037</td>
<td>9.9e-6</td>
<td>1.8e-06</td>
<td>Cacna2d3</td>
<td>decreased circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs13482117</td>
<td>14:27614362</td>
<td>HCT</td>
<td>HGB</td>
<td>0.023</td>
<td>0.03</td>
<td>5.9e-6</td>
<td>9.0e-07</td>
<td>Cacna2d3</td>
<td>decreased circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs6159786</td>
<td>14:27820736</td>
<td>HCT</td>
<td>HGB</td>
<td>0.026</td>
<td>0.037</td>
<td>9.9e-6</td>
<td>1.8e-06</td>
<td>Cacna2d3</td>
<td>decreased circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs6244569</td>
<td>14:27044891</td>
<td>HCT</td>
<td>HGB</td>
<td>0.026</td>
<td>0.037</td>
<td>9.9e-6</td>
<td>1.8e-06</td>
<td>Cacna2d3</td>
<td>decreased circulating glucose level</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>8</mn></msup><annotation encoding="application/x-tex">^8</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs13482288</td>
<td>14:81840412</td>
<td>ALY</td>
<td>BAS</td>
<td>0.036</td>
<td>0.65</td>
<td>1.78e-8</td>
<td>1.1e-07</td>
<td>Tdrd3</td>
<td>abnormal B cell differentiation and physiology</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>19</mn></msup><annotation encoding="application/x-tex">^{19}</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs3680448</td>
<td>14:81934085</td>
<td>ALY</td>
<td>BAS</td>
<td>0.036</td>
<td>0.65</td>
<td>1.78e-8</td>
<td>1.1e-07</td>
<td>Tdrd3</td>
<td>abnormal B cell differentiation and physiology</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>19</mn></msup><annotation encoding="application/x-tex">^{19}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs4173870</td>
<td>16:35764290</td>
<td>MCH</td>
<td>MCV</td>
<td>0.14</td>
<td>0.71</td>
<td>1.20e-11</td>
<td>4.89e-10</td>
<td>Hcls1</td>
<td>differentiation of erythrocytes</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>7</mn></msup><annotation encoding="application/x-tex">^7</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs4212102</td>
<td>16:84204704</td>
<td>PLT</td>
<td>WBC</td>
<td>0.17</td>
<td>0.35</td>
<td>1.16e-10</td>
<td>2.44e-9</td>
<td>App</td>
<td>increased susceptibility to induced thrombosis</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>20</mn><mo>,</mo><mn>11</mn></mrow></msup><annotation encoding="application/x-tex">^{20,11}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>rs4212186</td>
<td>16:84273330</td>
<td>PLT</td>
<td>WBC</td>
<td>0.17</td>
<td>0.36</td>
<td>5.88e-11</td>
<td>1.31e-9</td>
<td>App</td>
<td>increased susceptibility to induced thrombosis</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>20</mn><mo>,</mo><mn>11</mn></mrow></msup><annotation encoding="application/x-tex">^{20,11}</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td>rs3711994</td>
<td>19:45078018</td>
<td>ALY</td>
<td>LYM</td>
<td>3.71e-4</td>
<td>0.10</td>
<td>1.04e-12</td>
<td>2.80e-14</td>
<td>Btrc</td>
<td>abnormal lymphocyte morphology</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>21</mn></msup><annotation encoding="application/x-tex">^{21}</annotation></semantics></math></td>
</tr>
</tbody>
</table>
<p>In the first two columns, we list SNPs and their genetic location
according to the mouse assembly NCBI build 34 (accessed from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>21</mn></msup><annotation encoding="application/x-tex">^{21}</annotation></semantics></math>)
in the format <code>Chromosome:Basepair</code>. Next, we give the
results stemming from univariate analyses on traits 1 and 2,
respectively, the covariance (cov) test, and the overall
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>-value
derived by mvMAPIT using Fisher’s method. The last columns detail the
closest neighboring genes found using the Mouse Genome Informatics
database<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>4</mn></msup><annotation encoding="application/x-tex">^4</annotation></semantics></math><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>5</mn></msup><annotation encoding="application/x-tex">^5</annotation></semantics></math><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>6</mn></msup><annotation encoding="application/x-tex">^6</annotation></semantics></math>,
a short summary of the suggested annotated function for those genes, and
the reference to the source of the annotation.</p>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>1: William Valdar, Jonathan Flint, and Richard Mott. Simulating the
collaborative cross: power of quantitative trait loci detection and
mapping resolution in large sets of recombinant inbred strains of mice.
Genetics, 172(3):1783–1797, 2006. ISSN 0016-6731. <a href="https://doi.org/10.1534/genetics.104.039313" class="external-link uri">https://doi.org/10.1534/genetics.104.039313</a>.</p>
<p>2: William Valdar, Leah C. Solberg, Dominique Gauguier, Stephanie
Burnett, Paul Klenerman, William O. Cookson, Martin S. Taylor, J.
Nicholas P. Rawlins, Richard Mott, and Jonathan Flint. Genome-wide
genetic association of complex traits in heterogeneous stock mice.
Nature Genetics, 38(8):879–887, 2006. ISSN 1546-1718. <a href="https://doi.org/10.1038/ng1840" class="external-link uri">https://doi.org/10.1038/ng1840</a>.</p>
<p>3: Wellcome trust centre for human genetics - mouse resources. URL <a href="http://mtweb.cs.ucl.ac.uk/mus/www/mouse/index.shtml" class="external-link uri">http://mtweb.cs.ucl.ac.uk/mus/www/mouse/index.shtml</a>.</p>
<p>4: Judith A Blake, Richard Baldarelli, James A Kadin, Joel E
Richardson, Cynthia L Smith, and Carol J Bult. Mouse genome database
(MGD): Knowledgebase for mouse–human comparative biology. Nucleic Acids
Research, 49:D981–D987, 2020. ISSN 0305-1048. <a href="https://doi.org/10.1093/nar/gkaa1083" class="external-link uri">https://doi.org/10.1093/nar/gkaa1083</a>.</p>
<p>5: Cynthia L. Smith and Janan T. Eppig. The mammalian phenotype
ontology: enabling robust annotation and comparative analysis. Wiley
Interdisciplinary Reviews. Systems Biology and Medicine, 1(3):390–399,
2009. ISSN 1939-005X. <a href="https://doi.org/10.1002/wsbm.44" class="external-link uri">https://doi.org/10.1002/wsbm.44</a>.</p>
<p>6: Mouse Genome Informatics database <a href="http://www.informatics.jax.org" class="external-link uri">http://www.informatics.jax.org</a></p>
<p>7: Karla F. Castro-Ochoa, Idaira M. Guerrero-Fonseca, and Michael
Schnoor. Hematopoietic cell specific lyn substrate (HCLS1 or HS1): A
versatile actin-binding protein in leukocytes. Journal of Leukocyte
Biology, 105(5):881–890, 2019. ISSN 1938-3673. doi:
10.1002/JLB.MR0618-212R.</p>
<p>8: Obtaining and loading phenotype annotations from the international
mouse phenotyping consortium (IMPC) database. Mouse Genome Informatics
and the International Mouse Phenotyping Consortium, 2014. URL <a href="http://www.informatics.jax.org/reference/J:211773" class="external-link uri">http://www.informatics.jax.org/reference/J:211773</a>.</p>
<p>9: Marc H. G. P. Raaijmakers, Siddhartha Mukherjee, Shangqin Guo,
Siyi Zhang, Tatsuya Kobayashi, Jesse A. Schoonmaker, Benjamin L. Ebert,
Fatima Al-Shahrour, Robert P. Hasserjian, Edward O. Scadden, Zinmar
Aung, Marc Matza, Matthias Merkenschlager, Charles Lin, Johanna M.
Rommens, and David T. Scadden. Bone progenitor dysfunction induces
myelodysplasia and secondary leukaemia. Nature, 464(7290):852–857, 2010.
ISSN 1476-4687. doi: 10.1038/nature08851. URL <a href="https://www.nature.com/articles/nature08851" class="external-link uri">https://www.nature.com/articles/nature08851</a>. Number:
7290 Publisher: Nature Publishing Group.</p>
<p>10: Jill M. Norris and Stephen S. Rich. Genetics of glucose
homeostasis. Arteriosclerosis, Thrombosis, and Vascular Biology,
32(9):2091–2096, 2012. doi: 10.1161/ATVBAHA.112.255463. URL <a href="https://www.ahajournals.org/doi/10.1161/ATVBAHA.112.255463" class="external-link uri">https://www.ahajournals.org/doi/10.1161/ATVBAHA.112.255463</a>.
Publisher: American Heart Association.</p>
<p>11: Juan Carlos Souto, Laura Almasy, Montserrat Borrell, Francisco
Blanco-Vaca, José Mateo, José Manuel Soria, Inma Coll, Rosa Felices,
William Stone, Jordi Fontcuberta, and John Blangero. Genetic
susceptibility to thrombosis and its relationship to physiological risk
factors: The GAIT study. The American Journal of Human Genetics,
67(6):1452–1459, 2000. ISSN 0002-9297. doi: 10.1086/316903. URL <a href="https://doi.org/10.1086/316903" class="external-link uri">https://doi.org/10.1086/316903</a>.</p>
<p>12: Joachim Weischenfeldt, Inge Damgaard, David Bryder, Kim
Theilgaard-Mönch, Lina A. Thoren, Finn Cilius Nielsen, Sten Eirik W.
Jacobsen, Claus Nerlov, and Bo Torben Porse. NMD is essential for
hematopoietic stem and progenitor cells and for eliminating by-products
of programmed DNA rearrangements. Genes &amp; Development,
22(10):1381–1396, 2008. ISSN 0890-9369, 1549-5477. doi:
10.1101/gad.468808. URL <a href="https://genesdev.cshlp.org/content/22/10/1381" class="external-link uri">https://genesdev.cshlp.org/content/22/10/1381</a>. Company:
Cold Spring Harbor Laboratory Press Distributor: Cold Spring Harbor
Laboratory Press Institution: Cold Spring Harbor Laboratory Press Label:
Cold Spring Harbor Laboratory Press Publisher: Cold Spring Harbor
Lab.</p>
<p>13: Kevin Berendse, Maxim Boek, Marion Gijbels, Nicole N. Van der
Wel, Femke C. Klouwer, Marius A. van den Bergh-Weerman, Abhijit Babaji
Shinde, Rob Ofman, Bwee Tien Poll-The, Sander M. Houten, Myriam Baes,
Ronald J. A. Wanders, and Hans R. Waterham. Liver disease predominates
in a mouse model for mild human zellweger spectrum disorder. Biochimica
et Biophysica Acta (BBA) - Molecular Basis of Disease,
1865(10):2774–2787, 2019. ISSN 0925-4439. doi:
10.1016/j.bbadis.2019.06.013. URL <a href="https://doi.org/10.1016/j.bbadis.2019.06.013" class="external-link uri">https://doi.org/10.1016/j.bbadis.2019.06.013</a>.</p>
<p>14: Patricia Isnard, Nathalie Coré, Philippe Naquet, and Malek
Djabali. Altered lymphoid development in mice deficient for the mAF4
proto-oncogene. Blood, 96(2):705–710, 2000. ISSN 0006-4971. doi:
10.1182/blood.V96.2.705. URL <a href="https://doi.org/10.1182/blood.V96.2.705" class="external-link uri">https://doi.org/10.1182/blood.V96.2.705</a>.</p>
<p>15: Xiaoming Feng, Gregory C. Ippolito, Lifeng Tian, Karla Wiehagen,
Soyoung Oh, Arivazhagan Sambandam, Jessica Willen, Ralph M. Bunte,
Shanna D. Maika, June V. Harriss, Andrew J. Caton, Avinash Bhandoola,
Philip W. Tucker, and Hui Hu. Foxp1 is an essential transcriptional
regulator for the generation of quiescent naive t cells during thymocyte
development. Blood, 115(3):510–518, 2010. ISSN 0006-4971. doi:
10.1182/blood-2009-07-232694. URL <a href="https://doi.org/10.1182/blood-2009-07-232694" class="external-link uri">https://doi.org/10.1182/blood-2009-07-232694</a>.</p>
<p>16: 1451 Hui Hu, Bin Wang, Madhuri Borde, Julie Nardone, Shan Maika,
Laura Allred, Philip W. Tucker, and Anjana Rao. Foxp1 is an essential
transcriptional regulator of b cell development. Nature Immunology,
7(8):819–826, 2006. ISSN 1529-2916. doi: 10.1038/ni1358. URL <a href="https://www.nature.com/articles/ni1358" class="external-link uri">https://www.nature.com/articles/ni1358</a>. Number: 8
Publisher: Nature Publishing Group.</p>
<p>17: Monika Mortensen, Elizabeth J. Soilleux, Gordana Djordjevic,
Rebecca Tripp, Michael Lutteropp, Elham Sadighi-Akha, Amanda J. Stranks,
Julie Glanville, Samantha Knight, Sten-Eirik W. Jacobsen, Kamil R.
Kranc, and Anna Katharina Simon. The autophagy protein atg7 is essential
for hematopoietic stem cell maintenance. Journal of Experimental
Medicine, 208(3):455–467, 2011. ISSN 0022-1007. doi:
10.1084/jem.20101145. URL <a href="https://doi.org/10.1084/jem.20101145" class="external-link uri">https://doi.org/10.1084/jem.20101145</a>.</p>
<p>18: Alexander J. Clarke, Ursula Ellinghaus, Andrea Cortini, Amanda
Stranks, Anna Katharina Simon, Marina Botto, and Timothy J. Vyse.
Autophagy is activated in systemic lupus erythematosus and required for
plasmablast development. Annals of the Rheumatic Diseases,
74(5):912–920, 2015. ISSN 0003-4967, 1468-2060. doi:
10.1136/annrheumdis-2013-204343. URL <a href="https://ard.bmj.com/content/74/5/912" class="external-link uri">https://ard.bmj.com/content/74/5/912</a>. Publisher: BMJ
Publishing Group Ltd Section: Basic and translational research.</p>
<p>19: Yanzhong Yang, Kevin M. McBride, Sean Hensley, Yue Lu, Frederic
Chedin, and Mark T. Bedford. Arginine methylation facilitates the
recruitment of TOP3b to chromatin to prevent r loop accumu lation.
Molecular Cell, 53(3):484–497, 2014. ISSN 1097-2765. doi:
10.1016/j.molcel.2014.01.011. URL <a href="https://doi.org/10.1016/j.molcel.2014.01.011" class="external-link uri">https://doi.org/10.1016/j.molcel.2014.01.011</a>.</p>
<p>20: Feng Xu, Judianne Davis, Michael Hoos, andWilliam E. Van
Nostrand. Mutation of the kunitz-typeproteinase inhibitor domain in the
amyloid-protein precursor abolishes its anti-thrombotic properties in
vivo. Thrombosis Research, 155:58–64, 2017. ISSN 0049-3848. doi:
10.1016/j.thromres.2017.05.003. URL <a href="https://doi.org/10.1016/j.thromres.2017.05.003" class="external-link uri">https://doi.org/10.1016/j.thromres.2017.05.003</a>.</p>
<p>21: Keiko Nakayama, Shigetsugu Hatakeyama, Shun-ichiro Maruyama,
Akira Kikuchi, Kazunori Onoé, Robert A. Good, and Keiichi I. Nakayama.
Impaired degradation of inhibitory subunit of NF-b (IkappaB) and
-catenin as a result of targeted disruption of the -TrCP1 gene.
Proceedings of the National Academy of Sciences, 100(15):8752–8757,
2003. doi: 10.1073/pnas.1133216100. URL <a href="https://www.pnas.org/doi/full/10.1073/pnas.1133216100" class="external-link uri">https://www.pnas.org/doi/full/10.1073/pnas.1133216100</a>.
Publisher: Proceedings of the National Academy of Sciences.</p>
<p>22: Sagiv Shifman, 1480 Jordana Tzenova Bell, Richard R. Copley,
Martin S. Taylor, Robert W. Williams, Richard Mott, and Jonathan Flint.
A high-resolution single nucleotide polymorphism genetic map of the
mouse genome. PLOS Biology, 4(12):e395, 2006. ISSN 1545-7885. doi:
10.1371/journal. pbio.0040395. URL <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.0040395" class="external-link uri">https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.0040395</a>.
Publisher: Public Library of Science.</p>
<p>23: Stamp, Julian, 2022, “Leveraging the Genetic Correlation between
Traits Improves the Detection of Epistasis in Genome-wide Association
Studies”, <a href="https://doi.org/10.7910/DVN/WPFIGU" class="external-link uri">https://doi.org/10.7910/DVN/WPFIGU</a>, Harvard Dataverse,
V1</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
