<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Liability threshold MAPIT â€¢ mvMAPIT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Liability threshold MAPIT">
<meta property="og:description" content="Learn how to convert binary traits to liabilities.
">
<meta property="og:image" content="https://lcrawlab.github.io/mvMAPIT/logo.png">
<meta name="google-site-verification" content="7EMmOVQ0-Feh--5i1ogdQqVYz7Azip5Je-UeBYXP6fA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mvMAPIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/mvMAPIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">Tutorials</li>
    <li>
      <a href="../articles/tutorial-docker-mvmapit.html">Dockerized mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-lt-mapit.html">Liability threshold MAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-simulations.html">Simulate Traits</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Studies</li>
    <li>
      <a href="../articles/study-compare-p-value-combine-methods.html">Empirical comparison of P-value combination methods in mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/study-phillips-bnabs.html">Synergistic epistasis in binding affinity landscapes</a>
    </li>
    <li>
      <a href="../articles/study-wtccc-mice.html">Joint modeling of hematology traits yields epistatic signal in stock of mice</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lcrawlab/mvMAPIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><script src="tutorial-lt-mapit_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Liability threshold MAPIT</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/mvMAPIT/blob/main/vignettes/tutorial-lt-mapit.Rmd" class="external-link"><code>vignettes/tutorial-lt-mapit.Rmd</code></a></small>
      <div class="hidden name"><code>tutorial-lt-mapit.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/mvMAPIT" class="external-link">mvMAPIT</a></span><span class="op">)</span></span></code></pre></div>
<p>In this tutorial we will illustrate how to use the Liability Threshold MArginal ePIstasis Test (LT-MAPIT) by <a href="https://doi.org/10.1101/374983" class="external-link">Crawford and Zhou (2018)</a><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>. For this purpose we will first simulate synthetic data and then analyze it.</p>
<p>The data are single nucleotide polymorphisms (SNPs) with simulated genotypes. For the simulation we choose the following set of parameters:</p>
<ol style="list-style-type: decimal">
<li>
<strong>population_size</strong> - # of samples in the population</li>
<li>
<strong>n_snps</strong> - number of SNPs or variants</li>
<li>
<strong>PVE</strong> - phenotypic variance explained/broad-sense heritability (<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>H</mi><mn>2</mn></msup><annotation encoding="application/x-tex">H^2</annotation></semantics></math>)</li>
<li>
<strong>rho</strong> - measures the portion of <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>H</mi><mn>2</mn></msup><annotation encoding="application/x-tex">H^2</annotation></semantics></math> that is contributed by the marignal (additive) effects</li>
<li>
<strong>disease_prevalence</strong> - assumed disease prevelance in the population</li>
<li>
<strong>sample_size</strong> - # of samples to analyze</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">population_size</span> <span class="op">&lt;-</span> <span class="fl">1e4</span></span>
<span><span class="va">n_snps</span> <span class="op">&lt;-</span> <span class="fl">2e3</span></span>
<span><span class="va">pve</span> <span class="op">&lt;-</span> <span class="fl">0.6</span></span>
<span><span class="va">rho</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">disease_prevalence</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span><span class="va">sample_size</span> <span class="op">&lt;-</span> <span class="fl">500</span></span></code></pre></div>
<div class="section level3">
<h3 id="simulate-random-genotypes">Simulate random genotypes<a class="anchor" aria-label="anchor" href="#simulate-random-genotypes"></a>
</h3>
<p>Simulate the genotypes such that all variants have minor allele frequency (MAF) &gt; 0.05.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span> <span class="op">+</span> <span class="fl">0.45</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_snps</span><span class="op">)</span></span>
<span><span class="va">random_genotypes</span>   <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">population_size</span> <span class="op">*</span> <span class="va">n_snps</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">maf</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">population_size</span> <span class="op">*</span></span>
<span>                                                                         <span class="va">n_snps</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">maf</span><span class="op">)</span></span>
<span><span class="va">random_genotypes</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/double.html" class="external-link">as.double</a></span><span class="op">(</span><span class="va">random_genotypes</span><span class="op">)</span>, <span class="va">population_size</span>, <span class="va">n_snps</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="simulate-liabilities">Simulate liabilities<a class="anchor" aria-label="anchor" href="#simulate-liabilities"></a>
</h3>
<p>We can use the mvMAPIT function <code>simulate_traits</code> to simulate the liabilities. See the tutorial on simulations for more details on that.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_causal</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">n_epistatic</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">simulated_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_traits.html">simulate_traits</a></span><span class="op">(</span></span>
<span>    <span class="va">random_genotypes</span>,</span>
<span>    n_causal <span class="op">=</span> <span class="va">n_causal</span>,</span>
<span>    n_trait_specific <span class="op">=</span> <span class="va">n_epistatic</span>,</span>
<span>    n_pleiotropic <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    d <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    H2 <span class="op">=</span> <span class="va">pve</span>,</span>
<span>    rho <span class="op">=</span> <span class="va">rho</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Now that we have the liabilities, we can assign case-control labels according to the disease prevelance parameter. We will treat this like the LT-MAPIT paper and take an equal number of cases and controls.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">liabilities</span> <span class="op">&lt;-</span> <span class="va">simulated_data</span><span class="op">$</span><span class="va">trait</span></span>
<span><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">qnorm</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">disease_prevalence</span>, mean <span class="op">=</span> <span class="fl">0</span>, sd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">case_control_trait</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">population_size</span><span class="op">)</span></span>
<span><span class="va">case_control_trait</span><span class="op">[</span><span class="va">liabilities</span> <span class="op">&gt;</span> <span class="va">threshold</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="co"># Subsample a particular number of cases and controls</span></span>
<span><span class="va">cases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">liabilities</span> <span class="op">&gt;</span> <span class="va">threshold</span><span class="op">)</span>, <span class="va">sample_size</span> <span class="op">/</span> <span class="fl">2</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">controls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">liabilities</span> <span class="op">&lt;=</span> <span class="va">threshold</span><span class="op">)</span>, <span class="va">sample_size</span> <span class="op">/</span> <span class="fl">2</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">case_control_trait</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cases</span>, <span class="va">controls</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">X</span> <span class="op">&lt;-</span> <span class="va">simulated_data</span><span class="op">$</span><span class="va">genotype</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cases</span>, <span class="va">controls</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-mapit-with-case-control-trait">Run MAPIT with Case-Control trait<a class="anchor" aria-label="anchor" href="#run-mapit-with-case-control-trait"></a>
</h3>
<p>To run MAPIT with case-control traits, we need to convert the traits back to liabilities. The function <code>binary_to_liability</code> provides this conversion. <strong>NOTE:</strong> The <code>binary_to_liability</code> function is an approximation that is only suited for low prevalence in the disease trait.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y_liabilities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/binary_to_liability.html">binary_to_liability</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">disease_prevalence</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lt_mapit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mvmapit.html">mvmapit</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">y_liabilities</span><span class="op">)</span>,</span>
<span>        test <span class="op">=</span> <span class="st">"hybrid"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>1: Lorin Crawford, Xiang Zhou (2018) Genome-wide Marginal Epistatic Association Mapping in Case-Control Studies bioRxiv 374983; <a href="https://doi.org/10.1101/374983" class="external-link uri">https://doi.org/10.1101/374983</a></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
