<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Simulate Traits • mvMAPIT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Simulate Traits">
<meta property="og:description" content="Learn how to simulate traits with mvMAPIT.
">
<meta property="og:image" content="https://lcrawlab.github.io/mvMAPIT/logo.png">
<meta name="google-site-verification" content="7EMmOVQ0-Feh--5i1ogdQqVYz7Azip5Je-UeBYXP6fA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mvMAPIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/mvMAPIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">Tutorials</li>
    <li>
      <a href="../articles/tutorial-docker-mvmapit.html">Dockerized mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-lt-mapit.html">Liability threshold MAPIT</a>
    </li>
    <li>
      <a href="../articles/tutorial-simulations.html">Simulate Traits</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Studies</li>
    <li>
      <a href="../articles/study-compare-p-value-combine-methods.html">Empirical comparison of P-value combination methods in mvMAPIT</a>
    </li>
    <li>
      <a href="../articles/study-phillips-bnabs.html">Synergistic epistasis in binding affinity landscapes</a>
    </li>
    <li>
      <a href="../articles/study-wtccc-mice.html">Joint modeling of hematology traits yields epistatic signal in stock of mice</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lcrawlab/mvMAPIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><script src="tutorial-simulations_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Simulate Traits</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/mvMAPIT/blob/main/vignettes/tutorial-simulations.Rmd" class="external-link"><code>vignettes/tutorial-simulations.Rmd</code></a></small>
      <div class="hidden name"><code>tutorial-simulations.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/mvMAPIT" class="external-link">mvMAPIT</a></span><span class="op">)</span></span></code></pre></div>
<p>Data are simulated sample genotypes with <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math> single nucleotide polymorphisms (SNPs) and <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math> samples togeter with some quantitative traits.</p>
<div class="section level2">
<h2 id="simulate-genotypes">Simulate Genotypes<a class="anchor" aria-label="anchor" href="#simulate-genotypes"></a>
</h2>
<p>Create a matrix that contains underdetermined genotype data of random minor allele counts encoded as 0, 1, and 2. We control the minor allele frequency <code>maf</code> to be greater than 0.05.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span></span>
<span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">2938</span></span>
<span><span class="va">n_snp</span> <span class="op">&lt;-</span> <span class="fl">5747</span></span>
<span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span> <span class="op">+</span> <span class="fl">0.45</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_snp</span><span class="op">)</span></span>
<span><span class="va">random_minor_allele_counts</span>   <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_samples</span> <span class="op">*</span> <span class="va">n_snp</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">maf</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_samples</span> <span class="op">*</span> <span class="va">n_snp</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">maf</span><span class="op">)</span></span>
<span><span class="va">genotype_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="va">random_minor_allele_counts</span>,</span>
<span>  nrow <span class="op">=</span> <span class="va">n_samples</span>,</span>
<span>  ncol <span class="op">=</span> <span class="va">n_snp</span>,</span>
<span>  byrow <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">sample_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">n_samples</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span>fmt <span class="op">=</span> <span class="st">"id%04d"</span><span class="op">)</span></span>
<span><span class="va">snp_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">n_snp</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span>fmt <span class="op">=</span> <span class="st">"snp%04d"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">genotype_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">snp_names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">genotype_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_names</span></span></code></pre></div>
<p>The entries in the genotype matrix are:</p>
<ul>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0</mn><annotation encoding="application/x-tex">0</annotation></semantics></math> if the genotype is homozygous in the reference allele,</li>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math> if both, the reference allele and the alternative allele are present,</li>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>2</mn><annotation encoding="application/x-tex">2</annotation></semantics></math> if the genotype is homozygous in the alternative allele.</li>
</ul>
<p>An example genotype matrix could look like this:</p>
<table class="table">
<thead><tr class="header">
<th>Sample</th>
<th>SNP 1</th>
<th>SNP 2</th>
<th>…</th>
<th>SNP p</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0</td>
<td>1</td>
<td>…</td>
<td>2</td>
</tr>
<tr class="even">
<td>2</td>
<td>1</td>
<td>0</td>
<td>…</td>
<td>0</td>
</tr>
<tr class="odd">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>1</td>
</tr>
<tr class="even">
<td>n</td>
<td>2</td>
<td>2</td>
<td>…</td>
<td>1</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="simulate-traits">Simulate Traits<a class="anchor" aria-label="anchor" href="#simulate-traits"></a>
</h2>
<p>The simulated traits will be based on the genotype data simulated in the previous step. Simulations are implemented in the function <code>simulate_traits</code>. The causal SNPs as well as their effect sizes will be drawn at random. For simulating traits, we need to control the following simulation parameters.</p>
<p>Simulation Parameters:</p>
<ol style="list-style-type: decimal">
<li>
<code>n</code>: number of samples</li>
<li>
<code>p</code>: number of SNPs or variants</li>
<li>
<code>d</code>: number of traits</li>
<li>
<code>PVE</code>: phenotypic variance explained/broad-sense heritability (<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>H</mi><mn>2</mn></msup><annotation encoding="application/x-tex">H^2</annotation></semantics></math>)</li>
<li>
<code>rho</code>: portion of <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>H</mi><mn>2</mn></msup><annotation encoding="application/x-tex">H^2</annotation></semantics></math> that is contributed by the marginal (additive) effects</li>
<li>Number of SNPs that are causing variation in the trait:
<ul>
<li>
<code>n_causal</code>: total number of SNPs with any effects on traits.</li>
<li>
<code>n_trait_specific</code>: number of epistatic SNPs that affect a single trait.</li>
<li>
<code>n_pleiotropic</code>: number of epistatic SNPs that affect all traits.</li>
<li>
<code>group_ratio_pleiotropic</code>: ratio by which to split the epistatic SNPs into groups, e.g. 1:1, 1:3.</li>
</ul>
</li>
<li>
<code>epistatic_correlation</code>: correlation between the epistatic effects.</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seed</span> <span class="op">&lt;-</span> <span class="fl">67132</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="va">PVE</span> <span class="op">&lt;-</span> <span class="fl">0.6</span></span>
<span><span class="va">rho</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="va">n_causal</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span>
<span><span class="va">n_trait_specific</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">n_pleiotropic</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">group_ratio_pleiotropic</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">epistatic_correlation</span> <span class="op">&lt;-</span> <span class="fl">0.9</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">simulated_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_traits.html">simulate_traits</a></span><span class="op">(</span></span>
<span>    <span class="va">genotype_data</span>,</span>
<span>    n_causal <span class="op">=</span> <span class="va">n_causal</span>,</span>
<span>    n_trait_specific <span class="op">=</span> <span class="va">n_trait_specific</span>,</span>
<span>    n_pleiotropic <span class="op">=</span> <span class="va">n_pleiotropic</span>,</span>
<span>    d <span class="op">=</span> <span class="va">d</span>,</span>
<span>    H2 <span class="op">=</span> <span class="va">PVE</span>,</span>
<span>    rho <span class="op">=</span> <span class="va">rho</span>,</span>
<span>    epistatic_correlation <span class="op">=</span> <span class="va">epistatic_correlation</span>,</span>
<span>    group_ratio_pleiotropic <span class="op">=</span> <span class="va">group_ratio_pleiotropic</span>,</span>
<span>    maf_threshold <span class="op">=</span> <span class="va">maf</span>,</span>
<span>    seed <span class="op">=</span> <span class="va">seed</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We randomly choose causal variants that are split into three groups:</p>
<ol style="list-style-type: decimal">
<li>set of additive SNPs without interactions,</li>
<li>group 1 of interaction SNPs, and</li>
<li>group 2 of interaction SNPs.</li>
</ol>
<p>In the simulations carried out in this study, SNPs interact between sets. SNPs in the first group interact with SNPs in the second group, but do not interact with variants in their own group, and vice versa. One may view the SNPs in the first set as the “hubs” in an interaction map. We are reminded that interaction (epistatic) effects are different from additive effects. The following image illustrates the simulation of interactions.</p>
<div class="figure">
<img src="simulate_traits.png" style="width:80.0%" alt=""><p class="caption">Trait simulations. From all SNPs, a random subset of causal SNPs is sampled and assigned to one of three groups. SNPs in one of the two epistatic groups interact with all SNPs in the other epistatic group to simulate pairwise interactions.</p>
</div>
<blockquote>
<p><strong>NOTE</strong>: All causal SNPs in both the second and third groups have additive effects and are involved in pairwise interactions, while causal SNPs in the first set only have additive effects.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="details-on-simulating-pairwise-epistasis">Details on Simulating Pairwise Epistasis<a class="anchor" aria-label="anchor" href="#details-on-simulating-pairwise-epistasis"></a>
</h2>
<p>The general simulation scheme for interactions follows the methods outlined in Crawford et al. (2017). For the data representing the null hypothesis of no epistasis present, 1000 SNPs without rare variants are sampled from the genotype data. The effect sizes of the causal SNPs then are sampled independently from a multivariate normal distribution according to equation .</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mstyle mathvariant="bold"><mi>𝐛</mi></mstyle><mi>i</mi></msub><mo>∼</mo><msub><mstyle mathvariant="script"><mi>𝒩</mi></mstyle><mi>d</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><mstyle mathvariant="bold"><mi>𝐔</mi></mstyle><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\begin{equation}\label{eq:null_simulations_beta}
    \mathbf{b}_i \sim \mathcal{N}_{d}\left(0, \mathbf{U}\right)
\end{equation}</annotation></semantics></math></p>
<p>As parameter of the simulations, <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐔</mi></mstyle><annotation encoding="application/x-tex">\mathbf{U}</annotation></semantics></math> describes the covariance between the additive effects of the causal SNPs on the <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>d</mi><annotation encoding="application/x-tex">d</annotation></semantics></math> different traits. The effects of all other SNPs are set to zero.</p>
<p>In order to simulate pleiotropy, the number of desired pleiotropic SNPs is sampled from the set of causal SNPs. These SNPs are then included in simulating interactions for every trait. Additionally, trait specific SNPs are sampled independently for each trait from the causal SNPs without the pleiotropic SNPs.</p>
<p>The interactions are modeled by selecting the number of desired SNPs that have non-zero pairwise interaction effects. These SNPs then are split into two groups. Each SNP from one group is simulated to interact with each SNP of the other group but not with any SNP within the same group that it was assigned to. This results in <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>n</mi><mn>1</mn></msub><mo>⋅</mo><msub><mi>n</mi><mn>2</mn></msub></mrow><annotation encoding="application/x-tex">n_1 \cdot n_2</annotation></semantics></math> epistatic interactions, with <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mi>i</mi></msub><annotation encoding="application/x-tex">n_i</annotation></semantics></math> the number of SNPs in group i. For all these interactions, random effects are sampled from the multivariate normal distribution given in .</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mstyle mathvariant="bold"><mi>𝐚</mi></mstyle><mi>i</mi></msub><mo>∼</mo><msub><mstyle mathvariant="script"><mi>𝒩</mi></mstyle><mi>d</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><mstyle mathvariant="bold"><mi>𝐕</mi></mstyle><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\begin{equation}\label{eq:simulations_alpha}
    \mathbf{a}_i \sim \mathcal{N}_{d}\left(0, \mathbf{V}\right)
\end{equation}</annotation></semantics></math></p>
<p>Analogously, <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐕</mi></mstyle><annotation encoding="application/x-tex">\mathbf{V}</annotation></semantics></math> is the covariance between the pariwise epistatic effects of the causal SNPs on the <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>d</mi><annotation encoding="application/x-tex">d</annotation></semantics></math> different traits. The portion of the variance <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐄</mi></mstyle><annotation encoding="application/x-tex">\mathbf{E}</annotation></semantics></math> that is not due to heritable effects is simulated as multivariate normal with no correlation. The effects of the SNPs and interactions are collected in matrices <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐀</mi></mstyle><annotation encoding="application/x-tex">\mathbf{A}</annotation></semantics></math> and <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐁</mi></mstyle><annotation encoding="application/x-tex">\mathbf{B}</annotation></semantics></math>. The matrices <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐗</mi></mstyle><annotation encoding="application/x-tex">\mathbf{X}</annotation></semantics></math> and <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mstyle mathvariant="bold"><mi>𝐖</mi></mstyle><annotation encoding="application/x-tex">\mathbf{W}</annotation></semantics></math> contain the genotype data of the causal SNPs and interactions. With these effects, the phenotypes then are simulated according to equation .</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle><mo>=</mo><mstyle mathvariant="bold"><mi>𝐗</mi></mstyle><mstyle mathvariant="bold"><mi>𝐁</mi></mstyle><mo>+</mo><mstyle mathvariant="bold"><mi>𝐖</mi></mstyle><mstyle mathvariant="bold"><mi>𝐀</mi></mstyle><mo>+</mo><mstyle mathvariant="bold"><mi>𝐄</mi></mstyle></mrow><annotation encoding="application/x-tex">\begin{equation}\label{eq:simulation_mapit}
    \mathbf{Y} = \mathbf{X}\mathbf{B} + \mathbf{W}\mathbf{A} + \mathbf{E}
\end{equation}</annotation></semantics></math></p>
<p>For each trait independently, the variance components are scaled such that the variance can be partitioned according to equation .</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right"><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle></mtd><mtd columnalign="left"><mo>=</mo><msub><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle><mi>X</mi></msub><mo>+</mo><msub><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle><mi>W</mi></msub><mo>+</mo><mstyle mathvariant="bold"><mi>𝐄</mi></mstyle></mtd></mtr><mtr><mtd columnalign="right"><mstyle mathvariant="normal"><mi>v</mi><mi>a</mi><mi>r</mi></mstyle><mo stretchy="false" form="prefix">(</mo><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle><mo stretchy="false" form="postfix">)</mo></mtd><mtd columnalign="left"><mo>=</mo><mstyle mathvariant="normal"><mi>v</mi><mi>a</mi><mi>r</mi></mstyle><mo stretchy="false" form="prefix">(</mo><msub><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle><mi>X</mi></msub><mo stretchy="false" form="postfix">)</mo><mo>+</mo><mstyle mathvariant="normal"><mi>v</mi><mi>a</mi><mi>r</mi></mstyle><mo stretchy="false" form="prefix">(</mo><msub><mstyle mathvariant="bold"><mi>𝐘</mi></mstyle><mi>W</mi></msub><mo stretchy="false" form="postfix">)</mo><mo>+</mo><mstyle mathvariant="normal"><mi>v</mi><mi>a</mi><mi>r</mi></mstyle><mo stretchy="false" form="prefix">(</mo><mstyle mathvariant="bold"><mi>𝐄</mi></mstyle><mo stretchy="false" form="postfix">)</mo></mtd></mtr><mtr><mtd columnalign="right"></mtd><mtd columnalign="left"><mo>=</mo><mi>ρ</mi><mo>⋅</mo><msup><mi>H</mi><mn>2</mn></msup><mo>+</mo><mo stretchy="false" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>ρ</mi><mo stretchy="false" form="postfix">)</mo><mo>⋅</mo><msup><mi>H</mi><mn>2</mn></msup><mo>+</mo><mo stretchy="false" form="prefix">(</mo><mn>1</mn><mo>−</mo><msup><mi>H</mi><mn>2</mn></msup><mo stretchy="false" form="postfix">)</mo></mtd></mtr><mtr><mtd columnalign="right"></mtd><mtd columnalign="left"><mo>=</mo><mn>1</mn></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{align}\label{eq:simulation_var_comp}
    \mathbf{Y} &amp;= \mathbf{Y}_{X} + \mathbf{Y}_{W} + \mathbf{E} \\
    \mathrm{var}(\mathbf{Y}) &amp;= \mathrm{var}(\mathbf{Y}_{X}) +
                            \mathrm{var}(\mathbf{Y}_{W}) +
                            \mathrm{var}(\mathbf{E}) \\
                         &amp;= \rho \cdot H^2 + (1 - \rho) \cdot H^2 + (1 - H^2) \\
                         &amp;= 1
\end{align}</annotation></semantics></math></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
